benchmark_data_tools/rewrite_parquet.py

```diff
@@ -6,25 +6,36 @@
 from pathlib import Path
 from concurrent.futures import ThreadPoolExecutor
 
+# Perform a single-threaded pass to obtain the table names and schema.
+# Assumes hive data heirarchy.
+def map_table_schemas(input_dir, verbose):
```

Comment:
`duckdb_utils.py` is probably a better location for this function. Also, I don't think the directory path dependency is needed (i.e. DuckDB can return the table names).